test/language/statements/switch/syntax/redeclaration/async-function-declaration-attempt-to-redeclare-with-function-declaration.js

```diff
@@ -0,0 +1,20 @@
+// This file was procedurally generated from the following sources:
+// - src/declarations/redeclare-with-function-declaration.case
+// - src/declarations/redeclare-allow-sloppy-function/switch-attempt-to-redeclare-async-function-declaration.template
+/*---
+description: redeclaration with FunctionDeclaration (AsyncFunctionDeclaration in SwitchStatement)
+esid: sec-switch-statement-static-semantics-early-errors
+flags: [generated, async-functions]
+negative:
+  phase: early
+  type: SyntaxError
+info: |
+    SwitchStatement : switch ( Expression ) CaseBlock
+
+    It is a Syntax Error if the LexicallyDeclaredNames of CaseBlock contains any
+    duplicate entries.
+
+---*/
+
+
+switch (0) { case 1: async function f() {} default: function f() {} }
```

Comment:
In https://tc39.github.io/ecma262/#sec-switch-statement-static-semantics-lexicallydeclarednames, all lexically declared names from the different case clauses are combined. 
For each case clause, the LexicallyDeclaredNames definition from https://tc39.github.io/ecma262/#sec-block-static-semantics-lexicallydeclarednames applies, which has:

StatementListItem : Declaration
1. Return the BoundNames of Declaration. 

And in https://tc39.github.io/ecma262/#prod-Declaration we've got:
Declaration[Yield, Await] : HoistableDeclaration[?Yield, ?Await, ~Default]

And in https://tc39.github.io/ecma262/#prod-HoistableDeclaration:
HoistableDeclaration[Yield, Await, Default] : FunctionDeclaration[?Yield, ?Await, ?Default]

And the BoundNames of a FunctionDeclaration is its BindingIdentifier https://tc39.github.io/ecma262/#sec-function-definitions-static-semantics-boundnames.

And there's also B3.3.5 (https://tc39.github.io/ecma262/#sec-switch-duplicates-allowed-static-semantics) which allows duplicate FunctionDeclarations in sloppy mode in switch-statements.